compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -617,6 +613,14 @@ impl<'f> PerFunctionContext<'f> {
         }
     }
 
+    fn reduce_load_result_count(&mut self, value: ValueId) {
+        if let Some(context) = self.load_results.get_mut(&value) {
+            if context.uses != 0 {
```

Comment:
I added this to the load result decrement, even though it is not the ICE that @asterite was able to trigger. This should theoretically never underflow though if we are appropriately incrementing the counter, so perhaps we should remove the check here so that we can be notified about the load result usage counter being incremented or decremented incorrectly.